### PR TITLE
[JENKINS-67846] Icon alignment of build status is wrong

### PR DESCRIPTION
**Pull request by frankie139506**
See [JENKINS-67846](https://issues.jenkins-ci.org/browse/JENKINS-67846).
Add top: 0.2rem in `buildLink.jelly` to align icon vertically.
Add margin-left in `style.less` to add space between icon and text.

2022-02-23 23-18-03 的螢幕擷圖

### Proposed changelog entries

* Fix the position of icon and text (regression in 2.335).

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [X] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

**Comment by timja** [comment]
Was this fixed by https://github.com/jenkinsci/jenkins/pull/6341? cc @NotMyFault

File: core/src/main/resources/lib/hudson/buildLink.jelly
Comment (NotMyFault) [inline core/src/main/resources/lib/hudson/buildLink.jelly:55]:
```suggestion
          <l:icon alt="${r.iconColor.description}" class="${r.buildStatusIconClassName} icon-sm" style="margin-left: 0; position: relative; top: 0.2rem;"/><span class="jenkins-icon-adjacent">${jobName_}#<!-- -->${number}</span></a>
```

File: war/src/main/less/base/style.less
Comment (NotMyFault) [inline war/src/main/less/base/style.less:1578]:
```suggestion
```

**Review by NotMyFault — COMMENTED** [review]
Apologies for my late response, but I just got around it.

We likely can inline it and drop the root element on it, it's not needed here. But overall, it addresses the issue well.

**Review by NotMyFault — APPROVED** [review]
Should be good, other occasions I tested it in are unaffected by this change.

**Review by timja — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Comment by uhafner** [comment]
This is broken again, now vertically, see https://issues.jenkins.io/browse/JENKINS-67846.

**Pull request merged**
